python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -555,21 +581,89 @@ def get_repository_def(self) -> RepositoryDefinition:
         in order to access an functionality which is not exposed on Definitions. This method
         also resolves a PendingRepositoryDefinition to a RepositoryDefinition.
         """
+        inner_repository = self.get_inner_repository()
         return (
-            self._created_pending_or_normal_repo.compute_repository_definition()
-            if isinstance(self._created_pending_or_normal_repo, PendingRepositoryDefinition)
-            else self._created_pending_or_normal_repo
+            inner_repository.compute_repository_definition()
+            if isinstance(inner_repository, PendingRepositoryDefinition)
+            else inner_repository
         )
 
-    def get_inner_repository_for_loading_process(
+    @cached_method
+    def get_inner_repository(
         self,
     ) -> Union[RepositoryDefinition, PendingRepositoryDefinition]:
-        """This method is used internally to access the inner repository during the loading process
-        at CLI entry points. We explicitly do not want to resolve the pending repo because the entire
-        point is to defer that resolution until later.
+        """This method is used internally to access the inner repository. We explicitly do not want
+        to resolve the pending repo because the entire point is to defer that resolution until
+        later.
         """
-        return self._created_pending_or_normal_repo
+        return _create_repository_using_definitions_args(
+            name=SINGLETON_REPOSITORY_NAME,
+            assets=self.assets,
+            schedules=self.schedules,
+            sensors=self.sensors,
+            jobs=self.jobs,
+            resources=self.resources,
+            executor=self.executor,
+            loggers=self.loggers,
+            asset_checks=self.asset_checks,
+        )
 
     def get_asset_graph(self) -> AssetGraph:
         """Get the AssetGraph for this set of definitions."""
         return self.get_repository_def().asset_graph
+
+    def validate_loadable(self) -> None:
+        """Validates that the enclosed definitions will be loadable by Dagster:
+        - No assets have conflicting keys.
+        - No jobs, sensors, or schedules have conflicting names.
+        - All asset jobs can be resolved.
+        - All resource requirements are satisfied.
+
+        Raises an error if any of the above are not true.
+        """
+        self.get_inner_repository()
+
+    @staticmethod
+    def merge(*def_sets: "Definitions") -> "Definitions":
```

Comment:
I punted on thinking hard about whether this should be `*` args or a list. I don't have a strong opinion here.